environs/interface.go

```diff
@@ -327,3 +332,9 @@ type InstanceTagger interface {
 	// same names, but other existing tags will be left alone.
 	TagInstance(id instance.Id, tags map[string]string) error
 }
+
+// InstanceTypesFetcher is an interface that allows for instance information from
+// a provider to be obtained.
+type InstanceTypesFetcher interface {
```

Comment:
Is it really worth having a separate interface type for this single method? Why not just make it part of Environ?